Update PPP

Criss-Wang opened the pull request.

Criss-Wang committed:
Update deleteCommand activity diagram

Criss-Wang committed:
Update deleteCommand activity diagram

Criss-Wang committed:
Fix formating issue and typos

Criss-Wang committed:
Merge remote-tracking branch 'upstream/master' into finalize-dg

# Conflicts:
#	docs/DeveloperGuide.md
#	docs/images/DeleteCommandActivityDiagram.png

Criss-Wang committed:
Update PPP and fix DG errors

Criss-Wang committed:
Merge remote-tracking branch 'upstream/master' into finalize-dg

# Conflicts:
#	docs/DeveloperGuide.md

qlchan24 commented on docs/DeveloperGuide.md:
```diff
@@ -617,8 +617,11 @@ warehouse/supplier with the updated product. The method then updates the model w
 
-The following activity diagram summarizes what happens when a user updates a product:
-![Update Product Command Activity Diagram](images/UpdateCommandActivityDiagram.png)
+    The method then passes a `CommandResult` with a success message back to the `LogicManager`. Finally, the model
```
This is repeated from above

qlchan24 commented on docs/DeveloperGuide.md:
```diff
@@ -743,7 +758,7 @@ The list macros feature allows users to be able to view all presently saved macr
 #### How it is implemented
 
-The list macros feature is facilitated by the ListMacroCommand, whose job is to retrieve to list of macros and process it into a suitable format to be displayed to the user.
+The list macros feature is facilitated by the `ListMacroCommand`    , whose job is to retrieve to list of macros and process it into a suitable format to be displayed to the user.
```
The extra space seems unneccessary

qlchan24 commented on docs/DeveloperGuide.md:
```diff
@@ -687,8 +690,16 @@ If that is true, an exception will be thrown. This will be shown on the GUI as a
 
-The following activity diagram summarizes what happens when a user assigns a macro:
-![Assign Macro Command Activity Diagram](images/AssignMacroCommandActivityDiagram.png)
+    The `AssignMacroCommand#execute` method will then be called with the `model` instance. The method will first check if there is any existing macro in the model that uses the same alias.
```
This is also repeated above

qlchan24 commented on docs/DeveloperGuide.md:
```diff
@@ -732,8 +743,12 @@ retrieved macro will be removed from the model.
 
-The following activity diagram summarizes what happens when a user updates a product:
-![Remove Macro Command Activity Diagram](images/RemoveMacroCommandActivityDiagram.png)
+1. Result Display <br>
```
This is the same issue

qlchan24 reviewed (commented):
Some issues as stated, otherwise LGTM

jeffreytjs reviewed (commented):
Good pick up one the typos in DG. Do address Qin Liang's comments then it's good to go!

Criss-Wang committed:
Merge remote-tracking branch 'upstream/master' into finalize-dg

# Conflicts:
#	docs/DeveloperGuide.md

Criss-Wang committed:
A more detailed check on the content duplication issue

tohyuting commented on docs/DeveloperGuide.md:
```diff
@@ -543,11 +544,11 @@ In the following section, the interaction between different objects will be disc
    After receiving an input from user for edit command, `EditCommandParser#parse` will be invoked.
 
-   As mentioned in above section, if either one of the compulsory prefixes are missing, `ParseException` will be thrown to remind users. Furthermore, invalid values supplied for `type` and `index` (e.g. `i/test`), a `ParseException` will be thrown.
+   As mentioned in above section, if either one of the compulsory prefixes is missing, `ParseException` will be thrown to remind users. Furthermore, invalid values supplied for `type` and `index` (e.g. `String` value for `index`), a `ParseException` will be thrown.
```
I think this should be i/test instead, since we mentioned that String may not be clear :)

Criss-Wang replied:
Okay I think it's because I didn't check the content carefully, thanks!

tohyuting reviewed (commented):
Other than the issues pointed out my Qin Liang, just one more comment. Otherwise, LGTM.

Criss-Wang committed:
Fix merge conflict mistakes

Criss-Wang committed:
Fix more DG bugs with help from Yu Ting

Some diagram naming issues and missing/extra symbols are removed.
Problems with hyperlinks are fixed.

Criss-Wang committed:
Generate PNG images for PlantUML diagrams

Criss-Wang committed:
Add minor customization

qlchan24 reviewed (approved):
LGTM
